tests/Unit/Evolution/Initialization/Test_FunctionOfTimeFromFile.cpp

```diff
@@ -0,0 +1,138 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "tests/Unit/TestingFramework.hpp"
+
+#include <array>
+#include <cstddef>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "DataStructures/DataBox/DataBoxTag.hpp"
+#include "DataStructures/DataBox/Prefixes.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Domain/ElementId.hpp"
+#include "Domain/ElementIndex.hpp"
+#include "Domain/FunctionsOfTime/FunctionOfTime.hpp"
+#include "Domain/FunctionsOfTime/PiecewisePolynomial.hpp"
+#include "Domain/FunctionsOfTime/Tags.hpp"
+#include "Evolution/Initialization/FunctionOfTimeFromFile.hpp"
+#include "Evolution/Initialization/Tags.hpp"
+#include "Informer/InfoFromBuild.hpp"
+#include "Options/Options.hpp"
+#include "Options/ParseOptions.hpp"
+#include "Parallel/PhaseDependentActionList.hpp"
+#include "Utilities/FileSystem.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+#include "tests/Unit/ActionTesting.hpp"
+
+namespace {
+constexpr size_t dim = 2;
+using ElementIndexType = ElementIndex<dim>;
+
+template <typename Metavariables>
+struct component {
+  using metavariables = Metavariables;
+  using chare_type = ActionTesting::MockArrayChare;
+  using array_index = ElementIndexType;
```

Comment:
you can just use `size_t` as well since I don't think the action assumes `ElementIndex`

Reply:
When I tried this, I got these compiler errors. Changing back to ElementIndexType worked, so I'll leave it as that unless there's a fix you can suggest

~~~~~
/home/geoffrey/Codes/spectre/spectre/tests/Unit/Evolution/Initialization/Test_FunctionOfTimeFromFile.cpp:91:3: error: no matching function for call to 'emplace_component_and_initialize'
  ActionTesting::emplace_component_and_initialize<component<Metavariables>>(
  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/geoffrey/Codes/spectre/spectre/tests/Unit/ActionTesting.hpp:1623:6: note: candidate function not viable: no known conversion from 'const ElementId<dim>' to 'const typename component<Metavariables>::array_index' (aka 'const unsigned long') for 2nd argument
void emplace_component_and_initialize(
     ^
/home/geoffrey/Codes/spectre/spectre/tests/Unit/Evolution/Initialization/Test_FunctionOfTimeFromFile.cpp:100:10: error: no matching member function for call to 'next_action'
  runner.next_action<component<Metavariables>>(self_id);
  ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/geoffrey/Codes/spectre/spectre/tests/Unit/ActionTesting.hpp:1538:8: note: candidate function not viable: no known conversion from 'const ElementId<dim>' to 'const typename component<Metavariables>::array_index' (aka 'const unsigned long') for 1st argument
  void next_action(
       ^
/home/geoffrey/Codes/spectre/spectre/tests/Unit/Evolution/Initialization/Test_FunctionOfTimeFromFile.cpp:103:7: error: no matching function for call to 'get_databox_tag'
      ActionTesting::get_databox_tag<component<Metavariables>,
      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/geoffrey/Codes/spectre/spectre/tests/Unit/ActionTesting.hpp:1668:13: note: candidate function not viable: no known conversion from 'const ElementId<dim>' to 'const typename component<Metavariables>::array_index' (aka 'const unsigned long') for 2nd argument
const auto& get_databox_tag(
~~~~~

Reply:
try again and change the type of `self_id` below...